File: src/org/codepulsar/ast/statements/Print.java
package org.codepulsar.ast.statements;

import org.codepulsar.ast.Expression;
import org.codepulsar.ast.Statement;

public class Print extends Statement {
    private final Expression expression;

    private final int line;

    public Print(Expression expression, int line) {
        this.expression = expression;

        this.line = line;
    }

    public <R> R accept(Visitor<R> visitor) {
        return visitor.visitPrintExpression(this);
    }

    public Expression getExpression() {
        return this.expression;
    }

    public int getLine() {
        return this.line;
    }
}

File: src/codepulsar/ast/statement/Print.cpp
#include "Print.h"


Pulsar::Print::Print(Expression* expression, int line) {
    this->expression = expression;
    this->line = line;
}

std::any Pulsar::Print::accept(StmtVisitor& visitor) {
    return visitor.visitPrintStatement(this);
}

Pulsar::Expression* Pulsar::Print::getExpression() {
    return this->expression;
}

int Pulsar::Print::getLine() {
    return this->line;
}
